util/Migrator/DbScripts/2023-02-07_00_ProviderAddExistingOrganizations.sql

```diff
@@ -0,0 +1,40 @@
+CREATE PROCEDURE [dbo].[Organization_UnassignedToProviderSearch]
+    @Name NVARCHAR(50),
+    @OwnerEmail NVARCHAR(256),
+    @Skip INT = 0,
+    @Take INT = 25
+WITH RECOMPILE
+AS
+BEGIN
+    SET NOCOUNT ON
+    DECLARE @NameLikeSearch NVARCHAR(55) = '%' + @Name + '%'
+    DECLARE @OwnerEmailLikeSearch NVARCHAR(258) = '%' + @OwnerEmail + '%'
+
+SELECT
+    Q.*
+FROM (
+         SELECT DISTINCT
+             O.*,
+             (
+                 SELECT STRING_AGG(U.Email, ', ') WITHIN GROUP (ORDER BY U.Email ASC)
+         FROM
+             [dbo].[OrganizationUser] OU
+             INNER JOIN
+             [dbo].[User] U ON U.[Id] = OU.[UserId]
+         WHERE
+             OU.[OrganizationId] = O.Id
+           AND OU.[Type] = 0 --Get 'Owner' type users only
+         GROUP BY OU.[OrganizationId]
+     ) [OwnerEmails]
+FROM [dbo].[OrganizationView] O
+    ) Q
```

Comment:
This has 3 nested `SELECT` queries, which is hard to understand (at least for me) and probably bad for database performance. Can we do the same thing via `JOIN` instead, or generally simplify this query? Compare it to the Provider Search sproc, which has similar requirements (searching with 2 inputs): https://github.com/bitwarden/server/blob/master/src/Sql/dbo/Stored%20Procedures/Provider_Search.sql

Reply:
This was a bit complicated because I was returning the Owner emails but since that is not needed anymore I've simplified it.